addons/binding/org.openhab.binding.ikeatradfri/src/main/java/org/openhab/binding/ikeatradfri/handler/IkeaTradfriBulbHandler.java

```diff
@@ -152,41 +146,31 @@ public void onDataUpdate(JsonElement data) {
     }
 
     private void updateStatusFromProperties(LightProperties props) {
-        try {
-            OnOffType onoff = props.getOnOffState() ? OnOffType.ON : OnOffType.OFF;
-            updateState(CHANNEL_STATE, onoff);
-            logger.debug("Updating channel state: {}", onoff.toString());
+        if (!props.getOnOffState()) {
+            updateState(CHANNEL_BRIGHTNESS, PercentType.ZERO);
+            logger.debug("Updating channel brightness: {}", 0);
         }
-        catch(NullPointerException e) {}
 
-        try {
-            PercentType dimmer = props.getBrightness();
+        PercentType dimmer = props.getBrightness();
+        if (dimmer != null) {
             updateState(CHANNEL_BRIGHTNESS, dimmer);
             logger.debug("Updating channel brightness: {} from {}", dimmer.toString(), props.getBrightness());
         }
-        catch(NullPointerException e) {}
 
-        try {
-            PercentType colorTemp = props.getColorTemperature();
+        PercentType colorTemp = props.getColorTemperature();
```

Comment:
@kaikreuzer We get a NPE here if we have a thing of type `warmwhite_bulb`. That thing doesn't have the needed value for `TRADFRI_COLOR_X` in it's coap response. So we must skip the `getColorTemperature()` method in this case.
```
2017-05-25 21:01:49.460 [ERROR] [alifornium.core.network.CoapEndpoint] - Exception in protocol stage thread: null
java.lang.NullPointerException
	at org.openhab.binding.ikeatradfri.handler.IkeaTradfriBulbHandler$LightProperties.getColorTemperature(IkeaTradfriBulbHandler.java:111)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.openhab.binding.ikeatradfri.handler.IkeaTradfriBulbHandler.updateStatusFromProperties(IkeaTradfriBulbHandler.java:160)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.openhab.binding.ikeatradfri.handler.IkeaTradfriBulbHandler.onDataUpdate(IkeaTradfriBulbHandler.java:145)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.openhab.binding.ikeatradfri.handler.IkeaTradfriGatewayHandler$3.onLoad(IkeaTradfriGatewayHandler.java:264)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.CoapClient$ObserveMessageObserverImpl.deliver(CoapClient.java:1044)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.CoapClient$MessageObserverImpl.succeeded(CoapClient.java:970)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.CoapClient$MessageObserverImpl.onResponse(CoapClient.java:949)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.coap.Request.setResponse(Request.java:415)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.network.EndpointManager$ClientMessageDeliverer.deliverResponse(EndpointManager.java:249)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.network.stack.CoapStack$StackTopAdapter.receiveResponse(CoapStack.java:193)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.network.stack.AbstractLayer.receiveResponse(AbstractLayer.java:98)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.network.stack.ObserveLayer.receiveResponse(ObserveLayer.java:138)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.network.stack.AbstractLayer.receiveResponse(AbstractLayer.java:98)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.network.stack.BlockwiseLayer.receiveResponse(BlockwiseLayer.java:318)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.network.stack.AbstractLayer.receiveResponse(AbstractLayer.java:98)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.network.stack.ReliabilityLayer.receiveResponse(ReliabilityLayer.java:263)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.network.stack.AbstractLayer.receiveResponse(AbstractLayer.java:98)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.network.stack.CoapStack.receiveResponse(CoapStack.java:135)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.network.CoapEndpoint$InboxImpl.receiveMessage(CoapEndpoint.java:630)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.network.CoapEndpoint$InboxImpl.access$700(CoapEndpoint.java:536)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.network.CoapEndpoint$InboxImpl$1.run(CoapEndpoint.java:548)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at org.eclipse.californium.core.network.CoapEndpoint$5.run(CoapEndpoint.java:695)[375:org.openhab.binding.ikeatradfri:2.1.0.201705251821]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)[:1.8.0_65]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_65]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)[:1.8.0_65]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)[:1.8.0_65]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_65]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_65]
	at java.lang.Thread.run(Thread.java:745)[:1.8.0_65]
```